src/BlockTypes/AbstractBlock.php

```diff
@@ -94,6 +94,36 @@ public function enqueue_editor_assets() {
 		$this->enqueue_data();
 	}
 
+	/**
+	 * Injects block attributes into the block.
+	 *
+	 * @param string $content HTML content to inject into.
+	 * @param array  $attributes Key value pairs of attributes.
+	 * @return string Rendered block with data attributes.
+	 */
+	protected function inject_html_data_attributes( $content, array $attributes ) {
+		return preg_replace( '/<div /', '<div ' . $this->get_html_data_attributes( $attributes ), $content, 1 );
+	}
+
+	/**
+	 * Converts block attributes to HTML data attributes.
+	 *
+	 * @param array $attributes Key value pairs of attributes.
+	 * @return string Rendered HTML attributes.
+	 */
+	protected function get_html_data_attributes( array $attributes ) {
+		$data = [];
+
+		foreach ( $attributes as $key => $value ) {
+			if ( is_bool( $value ) ) {
+				$value = $value ? 'true' : 'false';
+			}
+			$data[] = 'data-' . esc_attr( strtolower( preg_replace( '/(?<!\ )[A-Z]/', '-$0', $key ) ) ) . '="' . esc_attr( $value ) . '"';
```

Comment:
Why is the `preg_replace` needed here? I'm assuming you want to handle something like this, `a KeyHere` so it becomes, `a-key-here` (where `esc_attr` takes care of removing the spaces)?